File: java/src/main/java/com/gaston/macbook/simplemvp/data/cache/ImageCacheImpl.java
package com.gaston.macbook.simplemvp.data.cache;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

public class ImageCacheImpl implements ImageCache {

    private SharedPreferences sharedPrefs;

    public ImageCacheImpl(Context context){
        sharedPrefs = PreferenceManager.getDefaultSharedPreferences(context);
    }

    @Override
    public void saveHdUrl(String hdurl) {
      sharedPrefs.edit().putString("hdurl", hdurl).apply();
    }
}

File: kotlin/src/main/java/com/gaston/macbook/kotlinsimplemvp/data/cache/ImageCacheImpl.kt
package com.gaston.macbook.kotlinsimplemvp.data.cache

import android.content.Context
import android.content.SharedPreferences
import android.preference.PreferenceManager

class ImageCacheImpl constructor(context:Context) : ImageCache {

    var sharedPrefs:SharedPreferences? = null

    init {
        sharedPrefs = PreferenceManager.getDefaultSharedPreferences(context)
    }

    override fun saveHdUrl(hdurl: String) {
        sharedPrefs?.edit()?.putString("hdurl",hdurl)?.apply()
    }
}